### PR TITLE
feat: allow resolving `.ts` files with `.js` extension

### LINKED ISSUE #96: Typescript imports with ".js" file extension are failing.
We should support importing `.ts` files via `.js`. 

Related issue:
https://github.com/vitejs/vite/issues/6671

Also, we should support `.mts` and `.cts` by default.

### PR DESCRIPTION
**Pull request by Akryum**
fix #96

### CODE REVIEW

File: src/jiti.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export default function createJITI(
     for (const ext of _additionalExts) {
       resolved =
         tryResolve(id + ext, options) ||
-        tryResolve(id.replace(/(c|m)?j(sx?)/, "$1t$2"), options) ||
+        tryResolve(id.replace(/(c|m)?j(sx?)$/, "$1t$2"), options) ||
```
Comment (pi0) [inline src/jiti.ts:178]: Can regex be more specific with dot ext?

Comment (Akryum) [reply src/jiti.ts]: Done

File: src/jiti.ts [hunk 2]
```diff
@@ -178,6 +175,7 @@ export default function createJITI(
     for (const ext of _additionalExts) {
       resolved =
         tryResolve(id + ext, options) ||
+        tryResolve(id.replace(/\.(c|m)?j(sx?)$/, ".$1t$2"), options) ||
```
Comment (pi0) [inline src/jiti.ts:178]:
```suggestion
        // TODO: Only do this for imports within a typescript parent module
        tryResolve(id.replace(/\.(c|m)?j(sx?)$/, ".$1t$2"), options) ||
```

**Review by pi0 — APPROVED** [review]
Thanks 💯

**Pull request merged**